### DISCUSSION TITLE
What are the minimum system requirements for Avalonia in Linux DRM?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#15059] [Q&A] [answered]
I plan to use Avalonia to develop some embedded devices. Due to the size of similar products and the price of the motherboard, I may need to use some small and low-performance motherboards, such as Raspberry Pi Zero or others.

Although I know that minimum system requirements are actually related to the amount of data processing, for example, higher performance is required for machine learning or AI operation.

But really all I need is to simply communicate with an Ethernet or serial port. So, I think for an application like mine, it's more Avalonia itself than my code that determines what motherboard I choose.

Therefore, can someone please tell me what are the approximate minimum system requirements for Avalonia to run under Linux DRM?

Thank you.

**Comment by thevortexcloud** [comment, marked as answer]
I have used a Pi 4b without issue before. An original Pi Zero however will not work as it's incapable of running .NET (the CPU architecture has no .NET support, although it can run Mono). The Pi Zero 2 however _should_ be able to do it, but I have never tried it so I can't comment on how well it works. I imagine the recently released Pi 5 will do great considering how much faster than it is than the Pi 4. 

If you want an actual minimum requirement list:

1. Needs to support .NET
2. Needs a DRM/framebuffer driver
3. Needs to support Skia (pretty much anything that can run .NET should support Skia)
4. Anecdotally: about ~300mb of RAM when running with a _very_  minimal Linux distro. Although you probably wont be doing much with that beyond running some of the sample apps. A more practical amount is ~512mb-1gb. Although I have not done extensive memory testing.

If it has all of those things it should theoretically work. Although how smooth it is probably depends on what you are doing, how you build the app and how you configure the OS.

**Reply by kyuranger** [reply]
Thanks very much!

**Reply by thevortexcloud** [reply]
Oh also I forgot to mention: If you need input, you need a touch screen. Mouse input wont work, and I don't think keyboards do either (although I have not tried it). You can find plenty of cheap 7" LCD touch screens around which are pretty good for this sort of thing. If you have access to a 3D printer, there are plenty of cases floating around that will fit the screen and a Pi.

**Reply by kyuranger** [reply]
I'm currently using a touch screen (I also made a virtual keyboard myself using Avalonia), but I'm not sure whether Avalonia Linux DRM supports mouse and keyboard (I haven't tried it yet). One of the more troublesome issues I encounter now is the casing. Most of the casings on the market are made of plastic and cannot be used in the industrial field. Although I found suppliers that can provide metal casings, the appearance is very ugly.

**Reply by thevortexcloud** [reply]
There plenty of metal cases for the Pi itself available, try looking at hobby electronics shops to get an idea as to what's available. The Argon NEO for example is just a simple metal box. There are a lot of fancier looking/weirder cases available too. But most of them are not designed for sitting in a factory as they lack things like air filters to keep dust out.  If you really want something unique/to put everything into a single unit you will probably need someone to custom make them for you, which likely wont be cheap.
